### PR TITLE
TPW Merger in Quarterly Reports

### PR DESCRIPTION
**Pull request by Charlie-Henry**
This PR adds support for the formerly PWD department in the bond dashboard. It accomplishes this by concatenating the department code and unit code in order to merge in the spend plan data stored in google drive. Two schema changes were needed.

This PR also includes some work addressing the comments in my past PR https://github.com/cityofaustin/atd-bond-reporting/pull/3.

issue https://github.com/cityofaustin/atd-data-tech/issues/12512

### CODE REVIEW

File: bond_data.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def to_postgres(client, df, table):
         params = {"select": "*", "updated_at": f"lt.{time}", "order": "updated_at"}
         res = client.delete(resource=table, params=params)
     except Exception as e:
-        raise e.text
+        raise e.response.text
```
Comment (Charlie-Henry) [inline bond_data.py:73]: @johnclary was right this was wrong

File: quarterly_reporting.py [hunk 2]
```diff
@@ -193,44 +200,75 @@ def main():
     soda = Socrata(SO_WEB, SO_TOKEN, username=SO_KEY, password=SO_SECRET, timeout=500, )
 
     # Download & transform subproject reports
-    subprojects = subproject_transformation(client)
-    df_to_socrata(soda, subprojects, SUBPROJECT_DATASET, True)
+    # subprojects = subproject_transformation(client)
+    # df_to_socrata(soda, subprojects, SUBPROJECT_DATASET, True)
```
Comment (Charlie-Henry) [inline quarterly_reporting.py:204]: Commenting these out because we are removing the subprojects tab for now

File: quarterly_reporting.py [hunk 3]
```diff
@@ -251,19 +290,32 @@ def main():
...
+    # Merging the spend plan into the expenses for previous quarter
     prev_exp = to_output(prev_exp, spend_plan)
+    # Creating column for selecting the quarter in Power BI
     prev_exp["quarter_selection"] = f"{prev_quarter[:4]} Q{prev_quarter[-1:]}"
 
+    # Append the previous expenses to the current expenses
     output = exp.append(prev_exp)
 
+    # Replace data in Socrata
+    output.drop(["department"], axis=1, inplace=True)
```
Comment (Charlie-Henry) [inline quarterly_reporting.py:318]: department data ends up being encoded in the `unit_code` column. Like: 2507D001. 2507 is ATD department and D001 is unit bikeways. Power BI needs this column in order to join the metadata table to the expenses table.

**Review by frankhereford — APPROVED** [review]
I've read over this code, and I think it looks great.

I had trouble building the image on my ARM mac, so I've put up a PR #4 to make it build on both Macs and on Intel, where it was building fine previously pm Intel, BTW. That PR shouldn't hold this up getting into `main` - it's low priority, and only really helpful for local testing on apple machines.

Would you like me to test this by running the script now that I have a local image? If so, can I get a `env_file` from you? No worries if that isn't helpful though.

Thanks again Charlie - 🚀🚀

**Comment by chiaberry** [comment]
I went to look for the env variables in 1Password and couldnt find them, but they are in prefect. Can I add them to 1password, or are they already there and I did a poor job searching @Charlie-Henry ?

**Review by mddilley — APPROVED** [review]
I did not test this code, but it looks great. Thanks for adding the comments - I think they are solid and will really help the next person who looks at this. 🙏 🚀

**Comment by Charlie-Henry** [comment]
> I went to look for the env variables in 1Password and couldnt find them, but they are in prefect. Can I add them to 1password, or are they already there and I did a poor job searching @Charlie-Henry ?

@chiaberry all secrets have been moved over to 1pass. Assume we'll migrate this over to airflow soon

**Comment by Charlie-Henry** [comment]
I'm going to go ahead and merge this in. @frankhereford has a PR open that'll allow this image to be built on ARM CPUs. I'll make some changes to the ETL to allow for the image to be upped to python 3.10 which is a requirement for the latest releases of mstrio-py.

**Pull request merged**